backend/app.py

```diff
@@ -58,13 +58,138 @@ def add_user():
+                  'id': exercise.id,
+                  'description': exercise.description,
+                  'duration': exercise.duration,
+                  'date': exercise.exercise_date
+                })
+
+        except Exception as e:
+            print(sys.exc_info(), e)
+            abort(422)
+
+    '''
+    GET /api/exercise/log
+    returns a json {'count': count} where count is the number of exercises
+
+    GET /api/exercise/log?user_id&from&to&limit
+    returns a json {'user': user, 'log': exercises} where user is the data of
+      the user, and log is the list of exercises registered by the user.
+      from, to, and limit are optional parameters
+    '''
+    @app.route('/api/exercise/log', methods=['GET'])
+    def retrieve_all_exercises():
+        user_id = request.args.get('user_id', None)
+        from_date = request.args.get('from', None)
+        to_date = request.args.get('to', None)
+        limit = request.args.get('limit', None, type=int)
+
+        try:
+            if not user_id:
+                exercises_counted = Exercise.query.count()
+
+                return jsonify({
+                  'count': exercises_counted
+                })
+            elif user_id:
+                user_data = User.query.filter(User.id == user_id).one_or_none()
+                user = user_data.format()
+                exercises_data = Exercise.query.filter(Exercise.user_uuid == user_id).all()
+
+                if from_date:
+                    exercises_data = Exercise.query.filter(
+                      Exercise.user_uuid == user_id,
+                      Exercise.exercise_date >= from_date
+                    ).all()
+                    if to_date:
+                        exercises_data = Exercise.query.filter(
+                          Exercise.exercise_date <= to_date,
+                          Exercise.exercise_date >= from_date,
+                          Exercise.exercise_date <= to_date
+                        ).all()
+
+                if to_date:
+                    exercises_data = Exercise.query.filter(
+                      Exercise.user_uuid == user_id,
+                      Exercise.exercise_date <= to_date
+                    ).all()
+                    if from_date:
+                        exercises_data = Exercise.query.filter(
+                          Exercise.exercise_date <= to_date,
+                          Exercise.exercise_date >= from_date,
+                          Exercise.exercise_date <= to_date
+                        ).all()
+
+                exercises = [exercise.short() for exercise in exercises_data]
+                if limit:
+                    exercises_data = exercises[0:limit]
+
+
+                if len(exercises_data):
+                    return jsonify({
+                      'user': user,
+                      'log': [exercises]
+                    })
+                else:
+                    abort(404)
```

Comment:
Should return an empty object. 404 errors are exclusive for when an endpoint/route/page does not exist.